cmd/kube-proxy/app/server_others.go

```diff
@@ -91,9 +91,40 @@ func newProxyServer(
 		return nil, fmt.Errorf("unable to register configz: %s", err)
 	}
 
+	hostname, err := utilnode.GetHostname(config.HostnameOverride)
+	if err != nil {
+		return nil, err
+	}
+
+	client, eventClient, err := createClients(config.ClientConnection, master)
+	if err != nil {
+		return nil, err
+	}
+
+	// kube-proxy needs to detect the IP family
+	// because iptables has a different binary per IP family
+
+	// The IP family will be determined by the nodeIP used by the proxier
+	// The order of precedence is:
+	// 1. config.bindAddress if bindAddress is not 0.0.0.0 or ::
+	// 2. based on the priority: NodeInternalIP, NodeExternalIP
+	//    the IP of the Node with the same nodename:
+	//  2.1 the provided hostname
+	//  2.2 the OS hostname
```

Comment:
You don't need to get into the details of how `utilnode.GetNodeIP` works. Just "`2. the primary IP from the Node object, if set`"

Reply:
done